#5228 Reconfigure cron job for view materialisation on staging server

tayowonibi opened the pull request:
re-configure bigquery views data materialisation cron job for the staging server  such that it does not materialise views on bigquery staging dataset in the bigquery production dataset by adding `--disable-view-name-mapping` flag

tayowonibi committed:
#5228

re-configure bigquery views data cron jobs for the staging server  sch that it does not materialize views on the production dataset

de-code commented on salt/data-pipeline/bigquery-views.sls:
```diff
@@ -11,10 +11,15 @@ re-materialise views daily:
 re-materialise views:
     cron.present:
         - user: {{ pillar.elife.deploy_user.username }}
+        {% if pillar.elife.env == "prod" %}
```
I think we usually put the environment specific configuration in the config that defines the properties, for each environment. Although `--disable-view-name-mapping` doesn't lend itself to it, so it's probably be okay this way. Will refer to @giorgiosironi for advice / thoughts.

tayowonibi replied:
thought of doing that , but the `pillar.elife.env`  is managed outside of this formula, and that is why i deferred to including the text directly in the cron job configuration

giorgiosironi replied:
1. Create `pillar.data_pipeline.bigquery_views.materialize_arguments` in https://github.com/elifesciences/data-pipeline-formula/blob/master/salt/pillar/data-pipeline.sls#L23
1. Add empty string value to https://github.com/elifesciences/builder-configuration/blob/master/pillar/data-pipeline-public.sls#L3
1. Add `--disable-view-name-mapping` value to https://github.com/elifesciences/builder-configuration/blob/master/pillar/environment-prod-public.sls (should be in the same PR, needs to add the parent keys in the YAML)
1. Change to use `{{pillar.data_pipeline.bigquery_views.materialize_arguments }}` removing the if

This way we can tune this by environment without changing the formula but only the configuration.

Separately: why `staging` shouldn't have the same behavior in principle?

giorgiosironi replied:
I can help with parts of this, will create the `builder-configuration` change.

tayowonibi replied:
> Separately: why staging shouldn't have the same behavior in principle?

@de-code : what do you think about having this? implication for the "view-mapping" functionality will be having `[ deployment_env_name ]_[ mapped_dataset_name ] ` as the eventual dataset name for mapped views ? if this is ok with you, should we implement this now, and leave this formula unmodified, or should we temporarily modify this formula to resolve the civi issue and then later implement this dataset name change functionality?

de-code replied:
> @de-code : what do you think about having this? implication for the "view-mapping" functionality will be having `[ deployment_env_name ]_[ mapped_dataset_name ] ` as the eventual dataset name for mapped views ? if this is ok with you, should we implement this now, and leave this formula unmodified, or should we temporarily modify this formula to resolve the civi issue and then later implement this dataset name change functionality?

I think in principal, doing the mapping for every deployment environment seems to make sense. That would however also affect the dataset name that the Civi intergration currently uses.

I think for now I would recommend doing the minimal to resolve the Civi issue and change the implementation later. (The only change I would maybe consider now is changing the default to avoid the exact same issue; EDIT: but even then it's probably best to first resolve the issue with the current implementation)

tayowonibi commented:
@giorgiosironi - is it possible for you to provide me with info on what needs to be done to fix the build failing issue?

giorgiosironi commented:
The build of `master` (`pr-base`) is broken as it says this file is a 404:
https://www-eu.apache.org/dist/nifi/nifi-registry/nifi-registry-0.4.0/nifi-registry-0.4.0-bin.tar.gz
It seems this has been removed in favor of https://www-eu.apache.org/dist/nifi/nifi-registry/nifi-registry-0.5.0/.

The only way to fix it seems to upgrade like it has been [done before](https://github.com/elifesciences/data-pipeline-formula/commit/709114c7d9fe7df7f885800241b6a07580c2a656#diff-195440acdea8c6fb810bec9ec0a7f70d). This needs to be a separate PR where we expect only `pr-fresh` to pass and will force the merge afterwards.

de-code commented:
> The build of `master` (`pr-base`) is broken as it says this file is a 404:
> https://www-eu.apache.org/dist/nifi/nifi-registry/nifi-registry-0.4.0/nifi-registry-0.4.0-bin.tar.gz
> It seems this has been removed in favor of https://www-eu.apache.org/dist/nifi/nifi-registry/nifi-registry-0.5.0/.

I was going to blame NiFi but that apache repo generally seem to contain the last version only. Should we maybe use http://archive.apache.org/dist/nifi/nifi-registry/ which has both the old and the new version?

tayowonibi committed:
#5228

re-configure bigquery views data cron jobs for the staging server  such that it does not materialize views on the production dataset

tayowonibi commented:
> 
> I was going to blame NiFi but that apache repo generally seem to contain the last version only. Should we maybe use http://archive.apache.org/dist/nifi/nifi-registry/ which has both the old and the new version?

i will create an issue and fix for that using your suggested solution

giorgiosironi commented on salt/pillar/data-pipeline.sls:
```diff
@@ -22,6 +22,7 @@ data_pipeline:
 
     bigquery_views:
         revision: latest
+        materialize_arguments:
```
I think you want an explicit empty string `""` as the default, otherwise this may gets printed as `None` when you use it

de-code replied:
actually, shouldn't the current default be `--disable-view-name-mapping`?

giorgiosironi replied:
Whatever you choose, we need to put it in https://github.com/elifesciences/builder-configuration/pull/95/files#diff-8e1abcc9fe49effc9fc073d53fdaa7edR25

tayowonibi replied:
> actually, shouldn't the current default be `--disable-view-name-mapping`?

current default is to use the mapping,  `--disable-view-name-mapping` needs to be  explicitly specified in order to use it

de-code replied:
> > actually, shouldn't the current default be `--disable-view-name-mapping`?
> 
> current default is to use the mapping, `--disable-view-name-mapping` needs to be explicitly specified in order to use it

What I meant is that currently, we only want to use the mapping in production (otherwise we'd end up with the issue we currently have). That is why in terms of deployment configuration, we should pass `--disable-view-name-mapping` by default, except for the `prod` configuration.

giorgiosironi commented:
https://github.com/elifesciences/builder-configuration/pull/95 will help.

giorgiosironi commented:
Also depends on https://github.com/elifesciences/data-pipeline-formula/pull/29

giorgiosironi commented:
Will need a merge of `master` after you have approved https://github.com/elifesciences/builder-configuration/pull/95 and I merged it. That will trigger a rebuild which should become green.

tayowonibi committed:
Merge branches 'fix_cron_job_for_staging_server' and 'master' of https://github.com/elifesciences/data-pipeline-formula into fix_cron_job_for_staging_server

tayowonibi committed:
#5228

re-configure bigquery views data cron jobs for the staging server  such that it does not materialize views on the production dataset

tayowonibi committed:
#5228

pass --disable-view-name-mapping as default configuration for view materialization

tayowonibi commented:
@giorgiosironi - can you help me debug this build. it keeps failing and i am not sure i know the reason for it

giorgiosironi commented:
This needed https://github.com/elifesciences/builder-configuration/pull/95 to be able to run.
